crates/ra_hir_def/src/find_path.rs

```diff
@@ -111,15 +111,40 @@ fn find_path_inner(
             None => continue,
             Some(path) => path,
         };
-        path.segments.push(name);
-        if path_len(&path) < best_path_len {
-            best_path_len = path_len(&path);
-            best_path = Some(path);
+        new_path.segments.push(name);
+
+        if prefer_new_path(best_path_len, best_path.as_ref(), &new_path) {
+            best_path_len = path_len(&new_path);
+            best_path = Some(new_path);
         }
     }
     best_path
 }
 
+fn prefer_new_path(old_path_len: usize, old_path: Option<&ModPath>, new_path: &ModPath) -> bool {
```

Comment:
Can we express this as a key function? like `(path.starts_with_std(), path.len())`?

Reply:
Yes, way more readable now, thanks.